recipes/moc/all/conanfile.py

```diff
@@ -0,0 +1,93 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from os import rename
+from os.path import isdir, join
+from shutil import move
+from glob import glob
+from distutils.version import LooseVersion
+
+class MocConan(ConanFile):
+    name = "moc"
+    homepage = "https://www.github.com/zuut/moc"
+    description = "Moc, the marked-up object compiler"
+    topics = ("conan", "moc", "idl", "code generation")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"    
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = { "fPIC": [True, False] }
+    default_options = { "fPIC": True }
+    _cmake = None
+    _supported_compilers = [
+        ("Linux", "gcc", "6"),
+        ("Linux", "clang", "6"),
+        ("Macos", "gcc", "6"),
+        ("Macos", "clang", "6"),
+        ("Macos", "apple-clang", "10")
+    ]
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if not self._check_compiler():
+            raise ConanInvalidConfiguration("%s package is not compatible with os %s and compiler %s version %s." % (self.name, self.settings.os, self.settings.compiler, self.settings.compiler.version))
+
+    def _check_compiler(self):
+        os = self.settings.os
+        compiler = self.settings.compiler
+        compiler_version = tools.Version(compiler.version)
+        return any(os == sc[0] and compiler == sc[1] and compiler_version >= sc[2] for sc in self._supported_compilers)
+
+    def build_requirements(self):
+        self.build_requires("flex/2.6.4")
+        self.build_requires("bison/3.7.1")
```

Comment:
```suggestion
    def requirements(self):
        self.requires("flex/2.6.4")
        self.requires("bison/3.7.1")
```

@intelligide suggested the correct syntax. Please accept suggestions instead of doing custom commits, it usually avoids mistakes and we can see who contributed to the recipe.